### ISSUE TITLE
Add E2E tests for TCPRoute

### ISSUE THREAD

**Issue by arkodg** [envoyproxy/gateway#2010]
*Description*:
>Describe the issue.

Add E2E tests in https://github.com/envoyproxy/gateway/tree/main/test/e2e/tests to test [TCPRoute](https://gateway-api.sigs.k8s.io/guides/tcp/)

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by arkodg** [comment]
thanks for picking this one up @slayer321, a suggestion would be to reuse the HTTP client and the HTTP server backend and attach a TCPRoute (maybe lets attach 2 TCPRoutes to 2 separate listeners) which should work just fine

**Comment by slayer321** [comment]
@arkodg After going through the code I found that there is no function available for `TCPRoute`  in the upstream gateway API like how we have it for [HTTPRoute](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/utils/kubernetes/helpers.go#L334) . So I wanted to ask do we first want to add it to upstream and then use it here . Or want to do it here and then add it to upstream.  

I tested as you mentioned with `2 TCPRoutes to 2 separate listeners` and it is working in both the scenarios(here by scenarios I mean I tested with making changes on gateway API as well as here). The Only issue I found is if we are creating `GatewayAndTCPRoutesMustBeAccepted` function here then we have to copy some of the other function as well which are not exported in the upstream (but anyway I think if we are making changes here first and then on the upstream. Then while making changes on the upstream we can remove those duplicate code form here.).

**Comment by arkodg** [comment]
@slayer321 check out https://github.com/envoyproxy/gateway/tree/main/test/e2e/tests, we apply YAML config per e2e test

**Comment by slayer321** [comment]
@arkodg Yeah, I already checked that. I have raised the [Draft PR](https://github.com/envoyproxy/gateway/pull/2109). Please take a look and let me know your thoughts.
